Sources/HederaTCK/JSONHelper.swift

```diff
@@ -0,0 +1,144 @@
+/*
+ * ‌
+ * Hedera Swift SDK
+ * ​
+ * Copyright (C) 2022 - 2024 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+internal func getJsonAsString(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> String {
+    guard let str = json.stringValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be a string.")
+    }
+    return str
+}
+
+internal func getJsonAsInt(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> Int64 {
+    guard let param = json.intValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be an integer.")
+    }
+    return param
+}
+
+internal func getJsonAsDouble(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> Double {
+    guard let param = json.doubleValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be a double.")
+    }
+    return param
+}
+
+internal func getJsonAsBoolean(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> Bool {
+    guard let param = json.boolValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be a boolean.")
+    }
+    return param
+}
+
+internal func getJsonAsList(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> [JSONObject] {
+    guard let param = json.listValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be a list.")
+    }
+    return param
+}
+
+internal func getJsonAsDict(_ json: JSONObject, _ paramName: String, _ functionName: String) throws -> [String:
+    JSONObject]
+{
+    guard let param = json.dictValue else {
+        throw JSONError.invalidParams("\(functionName): \(paramName) MUST be a dictionary.")
+    }
+    return param
+}
```

Comment:
Couldn't find a good way to template these functions here but if there is a way I would happily implement it

Reply:
@rwalworth Cleaner way may be to condense these into a single function that also accepts a custom `JsonType` enum, then uses a switch statement, and returns a generic value.

Reply:
Done